File: evo-ai-crm-community/lib/evo_extension_points/runtime_context.rb
# frozen_string_literal: true

module EvoExtensionPoints
  # Runtime context extension point. Community default: single-scope mode —
  # current_scope_id is always nil and with_scope is a pass-through. See
  # EXTENSION_POINTS.md.
  module RuntimeContext
    DEFAULT_CURRENT_ID = -> {}
    DEFAULT_WITH_SCOPE = ->(_id, &block) { block&.call }

    class << self
      def current_scope_id
        impl = EvoExtensionPoints.impl_for(:runtime_context_current_id) || DEFAULT_CURRENT_ID
        impl.call
      end

      def with_scope(id, &)
        impl = EvoExtensionPoints.impl_for(:runtime_context_with_scope) || DEFAULT_WITH_SCOPE
        impl.call(id, &)
      end
    end
  end
end

File: evo-ai-core-service-community/pkg/evoextensions/runtimecontext/runtimecontext.go
// Package runtimecontext is the public runtime-scope extension point of
// the community release. See EXTENSION_POINTS.md at the repository root.
package runtimecontext

import "context"

// Scope resolves the runtime scope identifier bound to a given request
// or background job. Implementations must be safe for concurrent use.
//
// The returned string is opaque to the community release; an empty
// string means "no scope bound", which is the standalone case.
type Scope interface {
	CurrentID(ctx context.Context) string
}

type noop struct{}

func (noop) CurrentID(context.Context) string { return "" }

// Default returns the no-op scope used when no extension is installed.
// It always reports the empty string, preserving the community
// release's single-scope behaviour.
func Default() Scope { return noop{} }
